src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
+
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  todo: Todo;
+  loadingTodos: number[];
+  handleRemoveTodo: (id: number) => void;
+  handleTodoStatusUpdate: (todo: Todo) => void;
+  handleTodoTitleUpdate: (todo: Todo, title: string) => Promise<void>;
+}
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  loadingTodos,
+  handleRemoveTodo,
+  handleTodoStatusUpdate,
+  handleTodoTitleUpdate,
+}) => {
+  const { id, title, completed } = todo;
+
+  const [newTitle, setNewTitle] = useState(title);
+  const [isDoubleClicked, setIsDoubleClicked] = useState(false);
+
+  const todoInputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (todoInputRef.current) {
+      todoInputRef.current.focus();
+    }
+  }, [isDoubleClicked]);
+
+  const handleDoubleClick = () => {
+    setIsDoubleClicked(true);
+  };
+
+  const handleTodoInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setNewTitle(event.target.value);
+  };
+
+  const saveChanges = () => {
+    handleTodoTitleUpdate(todo, newTitle);
+    setIsDoubleClicked(false);
+  };
+
+  const handleOnBlur = () => {
+    saveChanges();
+  };
```

Comment:
When I edit todo, remove the title and click outside of input - todo changes to completed and the title doesn't change.

![image](https://github.com/mate-academy/react_todo-app-with-api/assets/84296041/78173d19-a4aa-42be-b24b-8580c3265036)
![image](https://github.com/mate-academy/react_todo-app-with-api/assets/84296041/81e18ad4-2eb8-4b2a-b2bc-4d088bcc274e)

And after if I open edit mode - I don't have the title inside input
![image](https://github.com/mate-academy/react_todo-app-with-api/assets/84296041/60c2c186-d4c6-4e52-993c-2554be849ec5)